### ISSUE TITLE
Support different file systems for Extended Boot Loader Partition

### ISSUE THREAD

**Issue by paulmenzel** [systemd/systemd#14165]
systemd-gpt-auto-generator seems to assume, that the Extended Boot Loader Partition is always a VFAT file system, and mounting fails if it is formatted with ext4 for example.

    EXT4-fs (…): Unregcognized mount option "umask=0077" or missing value

But, the [specification][1] allows having more file systems.

> For systems where the firmware is able to read file systems directly, $BOOT must be a file system readable by the firmware. For other systems and generic installation and live media, $BOOT must be a VFAT (16 or 32) file system. Applications accessing $BOOT should hence not assume that fancier file system features such as symlinks, hardlinks, access control or case sensitivity are supported.

So, the assumption in the generator should be removed.

[1]: https://systemd.io/BOOT_LOADER_SPECIFICATION

**Comment by zdzichu** [comment]
So there are two options in specification:
- $BOOT readable by firmware: every system reads VFAT, some also read HPFS+. Does you firmware read ext4?
- other systems: $BOOT must be VFAT

So what is your situation?

**Comment by paulmenzel** [comment]
> So there are two options in specification:
> 
>     * $BOOT readable by firmware: every system reads VFAT, some also read HPFS+. Does you firmware read ext4?

Payloads are used with coreboot based firmware. GRUB can be used as a payload, and even the Linux kernel. Both come with drivers for all kinds of file systems, including ext4.

>     * other systems: $BOOT must be VFAT
> 
> So what is your situation?

I used a LUKS encrypted root partition (/), and for easy handling need an unencrypted `/boot` partition. There I used ext4 for to get all the advantages of permissions and so on. GRUB is boot loader on the EFI System Partition, and therefore it ext4 file systems can be read.
